.ci/watchdog/JenkinsWrapper.py

```diff
@@ -0,0 +1,82 @@
+# The source code contained or described herein and all documents related to the
+# source code ("Material") are owned by Intel Corporation or its suppliers or
+# licensors. Title to the Material remains with Intel Corporation or its
+# suppliers and licensors. The Material may contain trade secrets and proprietary
+# and confidential information of Intel Corporation and its suppliers and
+# licensors, and is protected by worldwide copyright and trade secret laws and
+# treaty provisions. No part of the Material may be used, copied, reproduced,
+# modified, published, uploaded, posted, transmitted, distributed, or disclosed
+# in any way without Intel's prior express written permission.
+# No license under any patent, copyright, trade secret or other intellectual
+# property right is granted to or conferred upon you by disclosure or delivery of
+# the Materials, either expressly, by implication, inducement, estoppel or
+# otherwise. Any license under such intellectual property rights must be express
+# and approved by Intel in writing.
+# Include any supplier copyright notices as supplier requires Intel to use.
+# Include supplier trademarks or logos as supplier requires Intel to use,
+# preceded by an asterisk. An asterisked footnote can be added as follows:
+# *Third Party trademarks are the property of their respective owners.
+# Unless otherwise agreed by Intel in writing, you may not remove or alter
+# this notice or any other notice embedded in Materials by Intel or Intel's
+# suppliers or licensors in any way.
+
+from time import sleep
+import requests
+import jenkins
+import logging
+
+# Logging
+log = logging.getLogger(__file__)
+ch = logging.StreamHandler()
+log.setLevel(logging.INFO)
+ch.setFormatter(logging.Formatter('%(name)s - %(levelname)s - %(message)s'))
+log.addHandler(ch)
+
+_RETRY_LIMIT = 3
+_RETRY_COOLDOWN = 15
+
+class JenkinsWrapper:
+    def __init__(self, jenkins_token, jenkins_user, jenkins_server):
+        self.jenkins_server = jenkins_server
+        self.jenkins = jenkins.Jenkins(jenkins_server, username=jenkins_user, password=jenkins_token)
+
+    def _try_jenkins(self, method, args=[]):
+        attempt = 0
+        while(attempt < _RETRY_LIMIT):
+            try:
+                return method(*args)
+            except Exception as e:
+                # Special case for get_queue_item
+                if "queue" in str(e) and "does not exist" in str(e):
+                    return {}
+                attempt = attempt + 1
+                log.warning("Failed to execute " + method.__name__ + " -- attempt: " + str(attempt) + " -- ERROR: " + str(e))
+            sleep(_RETRY_COOLDOWN)
+        raise RuntimeError("Unable to execute " + method.__name__ + " after " + str(_RETRY_LIMIT) + " retries.")
+
+    def get_build_console_output(self, job_name, build_number):
+        return self._try_jenkins(self.jenkins.get_build_console_output,[job_name, build_number])
+
+    def get_job_info(self, job_name):
```

Comment:
I would put `job_name` and `build_number` to `JenkinsWrapper` constructor and modify methods below to properties

Reply:
build_number is used multiple times during Watchdog execution (different CI job build for every PR).
job_name could be a property, but I'm using jenkins object to retrieve info about both CI job and Watchdog job. 

If I changed the methods below to properties I'd have to create separate jenkins objects for every job build. That'd mean a separate object for every PR + one for Watchdog job (because I'm attaching a link to Watchdog log in a message).

Reply:
ok